tuxemon/states/monster/__init__.py

```diff
@@ -262,3 +272,231 @@ def on_menu_selection_change(self) -> None:
             image = pygame.Surface((1, 1), pygame.SRCALPHA)
         self.monster_portrait.image = image
         self.refresh_menu_items()
+
+
+class JournalMenuState(Menu[Monster]):
+    """The journal menu allows you to view all monsters."""
+
+    background_filename = "gfx/ui/item/item_menu_bg.png"
+    draw_borders = False
+
+    def startup(self, **kwargs: Any) -> None:
+        self.state = "normal"
+
+        self.monster_sprite = Sprite()
+        self.sprites.add(self.monster_sprite, layer=100)
+
+        # do not move this line
+        super().startup(**kwargs)
+        self.menu_items.line_spacing = tools.scale(7)
+
+        # this is the area where the item description is displayed
+        rect = self.client.screen.get_rect()
+        rect.top = tools.scale(106)
+        rect.left = tools.scale(3)
+        rect.width = tools.scale(250)
+        rect.height = tools.scale(32)
+        self.text_area = TextArea(self.font, self.font_color, (96, 96, 128))
+        self.text_area.rect = rect
+        self.sprites.add(self.text_area, layer=100)
+
+        self.monster_center = self.rect.width * 0.16, self.rect.height * 0.45
+        self.monsters = db.lookup_entire_table("monster")
+        self.monster_index = 0
+        # The number of monsters to show in the list at once
+        self.monsters_to_show = 11
+
+    def calc_internal_rect(self) -> pygame.rect.Rect:
+        # area in the screen where the item list is
+        rect = self.rect.copy()
+        rect.width = int(rect.width * 0.58)
+        rect.left = int(self.rect.width * 0.365)
+        rect.top = int(rect.height * 0.05)
+        rect.height = int(self.rect.height * 0.60)
+        return rect
+
+    def on_menu_selection(self, menu_item: MenuItem[Monster]) -> None:
+        """
+        Called when player has selected a monster.
+
+        Currently, does nothing.
+
+        TODO: Open a new menu that shows monster details or something.
+
+        Parameters:
+            menu_item: Selected menu monster.
+
+        """
+        monster = menu_item.game_object
+
+    def initialize_items(self) -> Generator[MenuItem[Monster], None, None]:
+        """Refresh the visible tuxemon that will appear in the list.
+
+        We mostly want the cursor to stay where it is, and scroll the list of
+        monsters, except for the beginning or end of the list.
+
+        Therefore, scroll the list by collecting N Tuxemon that we want to
+        display right now, depending on the current monster_index.
+
+        TODO: Maybe make this more efficient in the future, rather than looping
+        through every single tuxemon.
+
+        :return: List(MenuItem)
```

Comment:
Wrong.

@bitcraft do we need this? typehints are enough

Reply:
This should be written in Google style ("Returns:" block), and explain the result semantics. As you said, ":type:" and ":rtype:" are redundant with type hints and should be removed.